### PR TITLE
fix: moving formula with multi cursor incorrectly shifts formula references

### LINKED ISSUE #1731: bug: moving formula with multi cursor incorrectly shifts formula references
Edit: this issue exists for copy-pasting formulas across sheets as well 

**To fix:** 
- [x] Dragging references 
- [x] Copy pasting references across sheets 

https://github.com/user-attachments/assets/c7d2bb32-0cc4-4f5e-b6b3-63cfb04929e3

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
closes #1731 

also cleaned up all clone operations

### CODE REVIEW

File: quadratic-core/src/controller/operations/clipboard.rs [hunk 1]
```diff
@@ -421,24 +426,33 @@ impl GridController {
...
                                 if matches!(code_cell.language, CodeCellLanguage::Formula) {
-                                    code_cell.code =
-                                        replace_internal_cell_references(&code_cell.code, dest_pos);
+                                    code_cell.code = replace_internal_cell_references(
+                                        &code_cell.code,
+                                        Pos {
+                                            x: insert_at.x + x as i64,
+                                            y: insert_at.y + y as i64,
+                                        },
+                                    );
```
Comment (AyushAgrawal-A2) [inline quadratic-core/src/controller/operations/clipboard.rs:455]: This solves the bug, rest of the diff is removing clone operations.

**Comment by luke-quadratic** [comment]
Fix works as intended

File: quadratic-core/src/controller/operations/clipboard.rs [hunk 2]
```diff
@@ -295,55 +297,58 @@ impl GridController {
...
-                        let mut border_selections = vec![];
-                        let mut border_styles = vec![];
-                        let sheet_rect: SheetRect = SheetPos {
-                            sheet_id: sheet.id,
-                            x: *x + start_pos.x,
-                            y: *y + start_pos.y,
+                clipboard
+                    .borders
+                    .into_iter()
+                    .for_each(|(x, y, cell_borders)| {
```
Comment (HactarCE) [inline quadratic-core/src/controller/operations/clipboard.rs:310]: nit: `for (x, y, cell_borders) in clipboard.borders {` would be more concise and indent less

**Review by ddimaria — APPROVED** [review]

**Review by HactarCE — CHANGES_REQUESTED** [review]
Good job reducing clones! Just one nitpick; otherwise LGTM

**Review by ddimaria — APPROVED** [review]

**Review by luke-quadratic — APPROVED** [review]

File: quadratic-core/src/controller/operations/clipboard.rs (hunk 2 above)
Comment (AyushAgrawal-A2) [reply quadratic-core/src/controller/operations/clipboard.rs]: thanks

**Review by HactarCE — APPROVED** [review]
LGTM!

**Pull request merged**